sh.ppy.osulazer.yaml

```diff
@@ -0,0 +1,99 @@
+command: osu-lazer
+modules:
+  - name: osu-lazer
+    buildsystem: simple
+    build-commands:
+      - dotnet publish osu.Desktop --source ./nuget-sources --framework net5.0 --configuration Release --runtime $RUNTIME --self-contained false --output /app/opt/ppy/osu-lazer /p:Version=2021.1028.0
+      - install.sh
+      - install -D -m 755 osu-lazer /app/bin/osu-lazer
+        # app icons
+      - install -D -m 644 lazer-512.png /app/share/icons/hicolor/512x512/apps/sh.ppy.osulazer.png
+      - install -D -m 644 lazer-256.png /app/share/icons/hicolor/256x256/apps/sh.ppy.osulazer.png
+      - install -D -m 644 lazer-128.png /app/share/icons/hicolor/128x128/apps/sh.ppy.osulazer.png
+      - install -D -m 644 lazer-64.png /app/share/icons/hicolor/64x64/apps/sh.ppy.osulazer.png
+        # mimetypes
+      - install -D -m 644 mime-512.png /app/share/icons/hicolor/512x512/mimetypes/sh.ppy.osulazer.x-osu-beatmap.png
+      - install -D -m 644 mime-256.png /app/share/icons/hicolor/256x256/mimetypes/sh.ppy.osulazer.x-osu-beatmap.png
+      - install -D -m 644 mime-128.png /app/share/icons/hicolor/128x128/mimetypes/sh.ppy.osulazer.x-osu-beatmap.png
+      - install -D -m 644 mime-64.png /app/share/icons/hicolor/64x64/mimetypes/sh.ppy.osulazer.x-osu-beatmap.png
+      - install -D -m 644 mime-48.png /app/share/icons/hicolor/48x48/mimetypes/sh.ppy.osulazer.x-osu-beatmap.png
+      - install -D -m 644 mime-32.png /app/share/icons/hicolor/32x32/mimetypes/sh.ppy.osulazer.x-osu-beatmap.png
+      - install -D -m 644 mime-16.png /app/share/icons/hicolor/16x16/mimetypes/sh.ppy.osulazer.x-osu-beatmap.png
+        # It seems that nautilus cannot display a MIME-type icon unless it is installed in the directory of the current theme.
+      - cp -a -x /app/share/icons/hicolor -T /app/share/icons/Adwaita
+      - cp -a -x /app/share/icons/hicolor -T /app/share/icons/breeze
+      - rm -r /app/share/icons/Adwaita/*/apps
+      - rm -r /app/share/icons/breeze/*/apps
+      - install -D -m 644 sh.ppy.osulazer.desktop /app/share/applications/sh.ppy.osulazer.desktop
+      - install -D -m 644 x-osu.xml /app/share/mime/packages/${FLATPAK_ID}.xml
+      - install -D -m 644 sh.ppy.osulazer.appdata.xml /app/share/metainfo/sh.ppy.osulazer.appdata.xml
+    build-options:
+        append-path: /usr/lib/sdk/dotnet5/bin
+        append-ld-library-path: /usr/lib/sdk/dotnet5/lib
+        arch:
+          aarch64:
+            env:
+              RUNTIME: linux-arm64
+          x86_64:
+            env:
+              RUNTIME: linux-x64
+        env:
+          PKG_CONFIG_PATH: /app/lib/pkgconfig:/app/share/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig:/usr/lib/sdk/dotnet5/lib/pkgconfig
+    sources:
+      - nuget.json
+      - type: git
+        url: https://github.com/peppy/osu.git
+        commit: a9208838a9dcff71a73fd0ab103f1251a2817d3e
+      - type: file
+        path: osu-lazer
+      - type: file
+        path: icons/lazer-512.png
```

Comment:
JFYI, unless also used internally, 512x512 icon would be automatically downscaled to smaller ones. Not sure about MIME icons.

Reply:
I don't know anything about the resize, but now I looked at my system, I don't have 512x512 application icons. Do you think it's worth deleting the one on 512?

Reply:
At the same time, icons of mime types on 512x512 exist. I think maybe it's for HI-DPI?